rhizome/host/lib/json_rpc_client.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require "json"
+require "socket"
+
+class JsonRpcError < StandardError
+  attr_reader :code
+
+  def initialize(message, code)
+    super(message)
+    @code = code
+  end
+end
+
+class JsonRpcClient
+  def initialize(socket_path, timeout = 5, response_size_limit = 1048576)
+    @socket_path = socket_path
+    @timeout = timeout
+    @response_size_limit = response_size_limit
+  end
+
+  def call(method, params = {})
+    # id is used to correlate the context between request and response.
+    # See https://www.jsonrpc.org/specification
+    id = rand(10000000)
```

Comment:
Does it need to be securerandom? I mean, on the unix socket, kinda whatever I guess?

Reply:
This is to correlate responses to requests if we had multiple calls in a single connection. Since we make a single request per connection at the moment, the value of `id` doesn't really add any information, and we could also use constant "1" instead of using random.

SPDK's rpc client implementation uses incrementing sequence numbers starting from 1.

Hence, this doesn't need to be secure random.